client/src/app.js

```diff
@@ -0,0 +1,207 @@
+import "./styles/style.css";
+import { API_KEY, BASE_URL, sources} from './variable';
+//Proxy usage to get all the Author names
+const authorNames = () => {
+    let validator = {
+        get(target, key) {
+            return key in target ? taget[key] : "No Author";
+        }
+      }
+    let authors = new Proxy({}, validator);
+}
+
+const settings = {
+  spinnerSelector: '.spinner',
+  ulSelector: '#newsArticlesList',
+  gridSection: '.newsSection',
+  channelUIListsSelector: '.thumbnail',
+  BASE_URL,
+  API_KEY
+}
+class Spinner {
+  constructor(selector) {
+    this.spinner = document.querySelector(selector)
+    this.spinner.style.display = 'none';
+  }
+
+  enableSpinner() {
+    this.spinner.style.display = 'block';
+  }
+
+  disableSpinner() {
+    this.spinner.style.display = 'none';
+  }
+
+}
+
+class App {
+  constructor (sources, settings) {
+    this.sources = sources
+    this.settings = settings
+    this.spinner = new Spinner(settings.spinnerSelector)
+    this.main = document.querySelector(settings.gridSection)
+    this.ul = document.querySelector(settings.ulSelector)
+    this.channelUILists = document.querySelector(settings.channelUIListsSelector)
+    //this.anchor = document.querySelector(settings.buttonSelector)
+    this.channels = {}
+  }
+
+  append(parent, el) {
+    return parent.appendChild(el)
+  }
+
+  createNode(element) {
+    return document.createElement(element)
+  }
+
+  initialContent() {
+    this.sources.forEach((item)=>{
+      this.main.innerHTML +=
+        `<div class="thumbnail">
+          <div class="caption">
+          <h3>${item.label}</h3>
+          <p><a href="#" class="btn btn-primary" data-load-channel="${item.inst}" role="button">${item.source}</a></p>
+          </div>
+        </div>`
+    }); 
+  }  
+
+  async get(source) {
+    this.spinner.enableSpinner()    
+      try {
+        const response = await fetch(`${BASE_URL}${source}&apiKey=${API_KEY}`);
+        return await response.json();
+      } catch (error) {
+        console.log(error);
+      }    
+    }   
+    
+  handleMainClick({ target }) {
+      if(target.tagName.toLowerCase() === 'a' && target.dataset.loadChannel) {
+        const channelName = target.dataset.loadChannel;
+        console.log(channelName);
```

Comment:
lets remove that console.log

Reply:
I kept it purposefully - As we have to remove the consoles from webpack.